### PR TITLE
adding File I/O tests to Fuzion's testing framework

### PR DESCRIPTION
**Pull request by WaelYoussfi**

### CODE REVIEW

**Comment by WaelYoussfi** [comment]
#541 #540 #539 #516 #515 need to be merged to carry out testing for both back-ends

File: tests/fileio/fileiotests.fz [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+
+  f := io.file
+
+  writer := f.write
+  reader := f.read
+  stats := f.stat
+
+  say "$dir exists: {exists dir}"
+  match writer.create_dir dir
+    _ unit => say "$dir was created"
```
Comment (fridis) [inline tests/fileio/fileiotests.fz:42]: the `match` syntax allows to just not give a variable name if you do not need it, so `_` can be removed.

**Review by fridis — COMMENTED** [review]
just a minor suggestion to remove redundant `_`.

**Comment by WaelYoussfi** [comment]
#582 needed for the new `stat.exists` to work

**Pull request merged**